### PR TITLE
Resources migration: Enabled migration tasks with batch processing.

### PR DESCRIPTION
**Pull request by KH-Amir-TN**
## Description

https://guides.rubyonrails.org/active_record_querying.html#retrieving-multiple-objects-in-batches

### CODE REVIEW

**Comment by KH-Amir-TN** [comment]
# DEPENDS ON #4009

File: lib/tasks/migrations/migrations.rake [hunk 1]
```diff
@@ -2,28 +2,31 @@
...
-        puts cyan "  Name: #{r.name}"
-      else
-        puts red "Unable to migrate Role:"
-        puts yellow "  ID: #{r.id}"
-        puts yellow "  Name: #{r.name}"
-        has_encountred_issue = 1 # At least one of the migrations failed.
-      end
+    Role.select(:id, :name)
+        .where.not(name: Role::RESERVED_ROLE_NAMES)
+        .find_each(batch_size: COMMON[:batch_size]) do |r|
```
Comment (farhatahmad) [inline lib/tasks/migrations/migrations.rake:15]: If we're going to use the default, I would drop the explicit `batch_size: ..` to explicitly show we're using the default value

Comment (KH-Amir-TN) [reply lib/tasks/migrations/migrations.rake:15]: I've added it to enable customization to suit admins needs without having to know about the internal of `ActiveRecord`,
You wanna me to remove it?

Comment (farhatahmad) [reply lib/tasks/migrations/migrations.rake:15]: ummm no that's a fair point. it's fine as is

**Pull request merged**